src/ZkFold/Symbolic/Data/Conditional.hs

```diff
@@ -17,3 +16,9 @@ class BoolType b => Conditional b a where
 
 instance {-# OVERLAPPABLE #-} (BoolType b, Eq b) => Conditional b x where
     bool f t b = if b == true then t else f
+
+find :: forall a t tt .
```

Comment:
Cam be generalized to `(Foldable f, Conditional b x) => f x -> (x -> b) -> Maybe x`